### DISCUSSION TITLE
Implementing an array of auto-complete fields using AlpineJS and a client-side list.

### DISCUSSION THREAD

**Discussion by instancezero** [livewire/livewire#5411] [1. Help]
I have a modal with a variable list of text fields, and I want to offer an auto-complete capability. Each field uses the same static list of several hundred elements for options, so I want to push that to the client just once.

I think I have some of this figured out but I'm not sure, and I can't figure out how to pass the ID of my local list back to the server (in the call to `pick()`).

Any assistance is greatly appreciated. This is a condensed version of what I have so far:

```
<form wire:submit.prevent="save">
    <script>
        document.addEventListener('alpine:init', () => {
            Alpine.store('titles', {
                // allNames is an array of {id, label} objects
                list: @js($allNames),

                selected: function (search) {
                    return this.list.filter(
                        i => i.startsWith(search)
                    );
                }
            })
        })
    </script>
    <x-modal.dialog wire:model.defer="showEditModal">
        <x-slot name="content">
            @foreach($fieldList as $key => $detail)
                <div wire:key="row-{{ $key }}">
                    <x-input.text wire:model="fieldList{{ $key }}"/>
                    <ul>
                        <template x-for="item in $store.titles.selected()">
                            <li wire:click="pick({{ $key }}, item.id)" x-text="item.label"></li>
                        </template>
                    </ul>
                </div>
            @endforeach
        </x-slot>
        // Submit, reset buttons....
    </x-modal.dialog>
</form>
```

**Comment by joshhanley** [comment]
@instancezero I would just use Livewire's javascript API to either call a method and pass parameters to it, or update a property. See here for details https://laravel-livewire.com/docs/2.x/inline-scripts#accessing-javascript-component-instance

Hope this helps!

**Comment by instancezero** [comment]
Thanks, Josh. I'll give it a try.

**Reply by joshhanley** [reply]
No worries! 🙂
